### DISCUSSION TITLE
Astro hybrid output response headers

### DISCUSSION THREAD

**Discussion by friedemannsommer** [withastro/docs#8739] [Ideas]
Is it the expected behavior of Astro's hybrid output to ignore the headers set in the `Response` object?

Suppose the following page exists:
```typescript
export const prerender = true

export async function GET(): Promise<Response> {
    return new Response('Hello World!', {
        status: 200,
        headers: {
            'content-type': 'text/plain; charset=utf-8'
        }
    })
}
```

My assumption would be that Astro would create a file containing the actual response body
and either separately or as part of the response body store the associated headers.

In the generated output a file containing the "Hello World!" string exists in the client directory,
but when requesting the route via an HTTP client, it doesn't send the associated `content-type` header.

<details>
<summary>Astro manifest</summary>

This is a simplified and stripped down example containing only the "prerender" route:
```json
{
    "adapterName": "@astrojs/node",
    "routes": [
        {
            "file": "prerender",
            "links": [],
            "scripts": [],
            "styles": [],
            "routeData": {
                "route": "/prerender",
                "isIndex": false,
                "type": "endpoint",
                "pattern": "^\\/prerender\\/?$",
                "segments": [
                    [
                        {
                            "content": "prerender",
                            "dynamic": false,
                            "spread": false
                        }
                    ]
                ],
                "params": [],
                "component": "src/pages/prerender.ts",
                "pathname": "/prerender",
                "prerender": true,
                "fallbackRoutes": [],
                "_meta": {
                    "trailingSlash": "ignore"
                }
            }
        }
    ],
    "base": "/",
    "trailingSlash": "ignore",
    "compressHTML": true,
    "componentMetadata": [],
    "renderers": [],
    "clientDirectives": [],
    "entryModules": {
        "/src/pages/prerender.ts": "chunks/prerender_CxqmRQwW.mjs"
    },
    "inlinedScripts": [],
    "assets": [
        "/prerender"
    ],
    "buildFormat": "directory",
    "checkOrigin": false,
    "rewritingEnabled": false,
    "experimentalEnvGetSecretEnabled": false
}
```
</details>

I have created a [StackBlitz example](https://stackblitz.com/edit/github-zy63jj?file=src%2Fpages%2Fprerender.ts) that includes the relevant settings and files.
Please run the following commands in order (top to bottom):
1. `npm run build`
2. `npm run preview`
3. Request `<preview URL>/prerender` with any HTTP client
4. Note the abscene of the `content-type: text/plain; charset=utf-8` header in the response

**Comment by delucis** [comment]
This is expected for prerendered routes. For anything that is pre-rendered Astro behaves essentially as a static site generator: it creates static HTML, CSS, etc. Deciding on MIME type headers etc. should then be done by whatever server ends up serving these static files. Usually this is done using a file extension — for example if you rename your example file to `src/pages/prerender.txt.ts`, it will generate a `prerender.txt` file and running `npm run preview` will show the correct `Content-Type` header.

**Reply by friedemannsommer** [reply]
Thanks, this clarified the behavior for me.
